### PR TITLE
Raster downloader: Overwrite outdated predictions

### PR DESCRIPTION
**Pull request by jdk24**
On new predictions:
1. Check for already existing rasters
2. If exists, overwrite raster file in file system
3. Delete and recreate the individual raster's table in PostGIS
4. Prevent duplicate insert into raster metadata (as raster is already published)

Add access to geoserver data for this
Print more meaningful messages on successful new insert / update

### CODE REVIEW

File: raster_download/src/main/java/de/meggsimum/sauber/sdi/RasterDownloader.java [hunk 1]
```diff
@@ -414,7 +413,7 @@ private void insertRaster(String fileName, String absPath) throws SQLException,I
 		conn.commit();
 		createSchema.close();
 		ProcessBuilder pb =
-				new ProcessBuilder("/bin/sh", "-c", "raster2pgsql -R -I -C -M -t auto "+ absPath +" "+  targetTable + " | PGPASSWORD="+ dbUserPw +" psql -h db -U "+ dbUser +" -d sauber_data -v ON_ERROR_STOP=ON");
+				new ProcessBuilder("/bin/sh", "-c", "raster2pgsql -d -R -I -C -M -t auto "+ absPath +" "+  targetTable + " | PGPASSWORD="+ dbUserPw +" psql -h db -U "+ dbUser +" -d sauber_data -v ON_ERROR_STOP=ON");
```
Comment (chrismayer) [inline raster_download/src/main/java/de/meggsimum/sauber/sdi/RasterDownloader.java:416]: Just for clarification: what does `-d`  for  `raster2pgsql` do?

File: raster_download/src/main/java/de/meggsimum/sauber/sdi/RasterDownloader.java [hunk 2]
```diff
@@ -425,13 +424,18 @@ private void insertRaster(String fileName, String absPath) throws SQLException,I
...
 		}
-	
-		try {
-			insertMetadata(absPath, conn);
-		} catch (SQLException e) {
-			System.out.println("Error inserting raster metadata into DB");
-			e.printStackTrace();
-		}	
+
+		if (rasterExists) {
```
Comment (chrismayer) [inline raster_download/src/main/java/de/meggsimum/sauber/sdi/RasterDownloader.java:428]: Shouldn't inserting of metadata be done if the raster did not exist?

**Review by chrismayer — COMMENTED** [review]
Thanks for this. LGTM in general, but I added a question you want to have a look at before merging.

File: raster_download/src/main/java/de/meggsimum/sauber/sdi/RasterDownloader.java (hunk 1 above)
Comment (jdk24) [reply raster_download/src/main/java/de/meggsimum/sauber/sdi/RasterDownloader.java:416]: So far we still import the raster both into the file system and database, the latter one being optional at this point. 
raster2pgsql creates a db table for each raster. To update outdated raster, `-d` deletes old table if exists.

**Review by chrismayer — APPROVED** [review]
LGTM now. Thanks for your ongoing effort on this

File: raster_download/src/main/java/de/meggsimum/sauber/sdi/RasterDownloader.java [hunk 3]
```diff
@@ -355,23 +355,22 @@ private void downloadRaster(String request, String fileName) throws IOException,
...
+
+			FileUtils.copyInputStreamToFile(is, imgFile);
 			String absPath = imgFile.getAbsolutePath();
 			System.out.println("Raster saved at " + absPath);
 			
 			is.close();
 
 			try {
-				insertRaster(fileName,absPath);  // insert raster into database via raster2pgsql
+				insertRaster(fileName,absPath,rasterExists);  // insert raster into database via raster2pgsql
```
Comment (JakobMiksch) [inline raster_download/src/main/java/de/meggsimum/sauber/sdi/RasterDownloader.java:373]: formatting: spaces between the commas and arguments

**Review by JakobMiksch — COMMENTED** [review]
just one formatting hint

**Pull request merged**